models/yang/sonic/sonic-nat.yang

```diff
@@ -0,0 +1,218 @@
+                        }
+                    }
+                }
+            }
+        }
+
+        container STATIC_NAT {
+            list STATIC_NAT_LIST {
+                key "global_ip";
+
+                leaf global_ip {
+                    type inet:ipv4-address;
+                }
+
+                leaf local_ip {
+                    mandatory true;
+                    type inet:ipv4-address;
+                }
+
+                leaf nat_type {
+                    type enumeration {
+                        enum SNAT;
+                        enum DNAT;
+                    }
+                    default DNAT;
+                }
+
+                leaf twice_nat_id {
+                    type uint16 {
+                        range "1..9999" {
+                            error-message "Invalid Group Id for twice NAT.";
+                        }
+                    }
+                }
+            }
+        }
+        container NAT_GLOBAL {
+            list NAT_GLOBAL_LIST {
+                key "key_leaf";
+                leaf key_leaf {
+                    type enumeration {
+                        enum Values;
+                    }
+                }
+                leaf admin_mode {
+                    type enumeration {
+                        enum ENABLE;
+                        enum DISABLE;
+                    }
+                }
+
+                leaf nat_timeout {
+                    type uint32 {
+                        range "300..432000" {
+                            error-message "Invalid NAT Timeout, valid range 300 sec to 432000 sec.";
+                        }
+                    }
+                }
+                leaf nat_tcp_timeout {
+                    type uint32 {
+                        range "300..432000" {
+                            error-message "Invalid NAT TCP Timeout, valid range 300 sec to 432000 sec.";
+                        }
+                    }
+                }
+                leaf nat_udp_timeout {
+                    type uint16 {
+                        range "120..600" {
+                            error-message "Invalid NAT UDP Timeout, valid range 120 sec to 600 sec.";
+                        }
+                    }
+                }
+            }
+        }
+        container NAT_POOL {
+            list NAT_POOL_LIST {
+                key "pool_name";
+
+                leaf pool_name {
+                    type string;
+                }
+
+                choice nat_ip {
+                    case ip_address {
+                        leaf IP_ADDRESS {
+                            type inet:ipv4-address;
+                        }
+                    }
+                    case ip_address_range {
+                        leaf IP_ADDRESS_RANGE {
+                            type string {
+                                pattern '^(([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|'  +
```

Comment:
Please check whether ietf-inet has similar IP address range support and use it accordingly. 
We don't need '^' and '$' at the end of pattern. It would result in error while matching.

Reply:
As per the comment removed the extra characters. Will explore any pre defined type available in ietf yangs for ip address range.